### PR TITLE
Added soql-null

### PR DESCRIPTION
**Pull request by rozap**
* Added soql-null
  * added a new smoke test with a bunch of nulls
  * fixed broken behavior where null values would
    trigger creation of a new layer
* updated test workflow
* updated readme, remove babel-node note
* Made spill to disk param configurable

### CODE REVIEW

File: lib/decoders/layer.js [hunk 1]
```diff
@@ -115,18 +109,86 @@ class Layer extends Duplex {
...
+   * are eliminated in the shapefile library being used.
+   *
+   * This could be written as a big map, but it is called on every write,
+   * so it's important to return as early as possible.
+   *
+   * @mutates this.columns
+   */
+  _updateColumnTypes(soqlRow) {
+    var undefinedColumns = this.columns.filter((c) => c.ctype === 'null');
+    if (!undefinedColumns.length) return;
```
Comment (catstavi) [inline lib/decoders/layer.js]: This was a little hard to follow as someone who didn't intuit that 0 casts to false in javascript.

File: lib/decoders/layer.js [hunk 2]
```diff
@@ -115,18 +109,86 @@ class Layer extends Duplex {
...
 
+  /**
+   * Write the soqlRow with a given crs to the layer.
+   * This will only be called if belongsIn returned true for the soqlRow and this layer.
+   * See the note above on the confusing mutate-y  bits.
+   * @param  {[type]} crs       [description]
+   * @param  {[type]} soqlRow   [description]
+   * @param  {[type]} throwaway [description]
+   * @return {[type]}           [description]
+   */
```
Comment (catstavi) [inline lib/decoders/layer.js]: the documentation so far on this pr has been so, so good it's a little disappointing to not get a description of what crs is!

**Comment by catstavi** [comment]
It would be cool to see the documentation for write to be finished, other than that, looks good! :+1:

File: lib/decoders/layer.js (hunk 1 above)
Comment (rozap) [inline lib/decoders/layer.js]: sounds good, i'll make these explicit
